### ISSUE TITLE
Make an additional optional feature to use `deflate` and `inflate` instead of `flate2`

### ISSUE THREAD

**Issue by application-developer-DA** [mvdnes/zip-rs#60]
I think it makes sense to provide some additional feature-guarded backends / "deflate engines" to the end user, currently `flate2` is used, but the problem with it is that it relies on some C libraries and even the `rust_backend` pulls too much C-related things and requires a refactoring. When I tried to port `zip-rs` to `WebAssembly` (Rust now has a target which allows you to compile the library as WebAssembly module) I struggled with `flate2` and its dependencies (notably it's `rust_backend` dependency) and had to replace the usage of `flate2` by the combination of `deflate` and `inflate` (that was the easiest way to go), both provide pure Rust libraries for compression and decompression. The API of `inflate` is very similar to the one provided by `flate2`, so the changes were trivial, `deflate` has however a different API and has to be adapted a bit to be able to work with `zip-rs`.

IMHO it also makes sense to enable/disable compression and decompression seperately (some people may only either compression or decompression, which is especially important for things like `WebAssembly` where you don't want to blow up the size of the compiled module).

**Comment by xmclark** [comment]
@application-developer-DA did you make a fork with the refactor for `deflate` and `inflate`? I also am interested in using `zip-rs` in WebAssembly, but have not had much luck.

**Comment by application-developer-DA** [comment]
@xmclark yes, I did some changes several months ago. I did not make a pull request back then, since the changes I did are not streamlined yet and it was kind of an "experimental" version. Maybe I find time some day and update the fork: https://github.com/snapview/zip-rs/tree/wasm-experiments

The fork works fine with WebAssembly. I was able to compile and use it several months ago and the compiled WASM module is still used within the project for which I did this "small experiment", I have not had any issues with it so far.
